### PR TITLE
[SYCL] Fix image::get_size method

### PR DESCRIPTION
**Pull request by romanovvlad**
https://github.com/intel/llvm/pull/6600 changed behavior of  sycl::image::get_size() to return number of elements rather
than number of bytes which an image uses. This is not correct.
Changing it back and rename the internal method so it's
more clear about what it returns.

### CODE REVIEW

**Comment by aelovikov-intel** [comment]
I think some description in the PR is needed. It's not immediately clear from the diff what was the issue and why the fix isn't NFC. To me it looked like changing names without changing what the functions do.

**Comment by romanovvlad** [comment]
> I think some description in the PR is needed. It's not immediately clear from the diff what was the issue and why the fix isn't NFC. To me it looked like changing names without changing what the functions do.

Added.

File: sycl/source/detail/plugin.hpp [hunk 1]
```diff
@@ -174,6 +174,7 @@ class plugin {
     // If arguments need to be captured, then a data structure can be sent in
     // the per_instance_user_data field.
     const char *PIFnName = PiCallInfo.getFuncName();
+    printf("%s\n", PIFnName);
```
[the remark this answers is not in the document]
Comment (romanovvlad) [reply sycl/source/detail/plugin.hpp]: Thanks.Removed.

File: sycl/source/buffer.cpp [hunk 2]
```diff
@@ -119,7 +119,7 @@ void buffer_plain::addOrReplaceAccessorProperties(
   impl->addOrReplaceAccessorProperties(PropertyList);
 }
 
-size_t buffer_plain::getSize() const { return impl->getSize(); }
+size_t buffer_plain::getSize() const { return impl->getSizeInBytes(); }
```
Comment (aelovikov-intel) [inline sycl/source/buffer.cpp:122]: How is this related to the image?

Comment (aelovikov-intel) [reply sycl/source/buffer.cpp:122]: Resolved offline - my misunderstanding.

**Review by aelovikov-intel — APPROVED** [review]
